qiskit_experiments/framework/backend_data.py

```diff
@@ -0,0 +1,145 @@
+        if isinstance(backend, BackendV1):
+            return backend.configuration().control(qubits)
+        elif isinstance(backend, BackendV2):
+            return backend.control_channel(qubits)
+        return None
+
+    @staticmethod
+    def granularity(backend):
+        """Returns the backend's time constraint granularity"""
+        if isinstance(backend, BackendV1):
+            return backend.configuration().timing_constraints.get("granularity", None)
+        elif isinstance(backend, BackendV2):
+            return backend.target.granularity
+        return None
+
+    @staticmethod
+    def set_granularity(backend, value):
+        """Sets the backend's time constraint granularity"""
+        if isinstance(backend, BackendV1):
+            setattr(
+                backend.configuration(),
+                "timing_constraints",
+                {"granularity": value},
+            )
+        elif isinstance(backend, BackendV2):
+            backend.target.granularity = 16
+
+    @staticmethod
+    def dt(backend):
+        """Returns the backend's input time resolution"""
+        if isinstance(backend, BackendV1):
+            return backend.configuration().dt
+        elif isinstance(backend, BackendV2):
+            return backend.dt
+        return None
+
+    @staticmethod
+    def max_experiments(backend):
+        """Returns the backend's max experiments value"""
+        if isinstance(backend, BackendV1):
+            return getattr(backend.configuration(), "max_experiments", None)
+        elif isinstance(backend, BackendV2):
+            return backend.max_circuits
+        return None
+
+    @staticmethod
+    def coupling_map(backend):
+        """Returns the backend's coupling map"""
+        if isinstance(backend, BackendV1):
+            return getattr(backend.configuration(), "coupling_map", [])
+        elif isinstance(backend, BackendV2):
+            return list(backend.coupling_map.get_edges())
+        return []
+
+    @staticmethod
+    def control_channels(backend):
+        """Returns the backend's control channels"""
+        if isinstance(backend, BackendV1):
+            return getattr(backend.configuration(), "control_channels", None)
+        elif isinstance(backend, BackendV2):
+            return backend.control_channels
+        return None
+
+    @staticmethod
+    def version(backend):
+        """Returns the backend's version"""
+        if isinstance(backend, BackendV1):
+            return getattr(backend, "version", None)
+        elif isinstance(backend, BackendV2):
+            return backend.version
+        return None
+
+    @staticmethod
+    def provider(backend):
+        """Returns the backend's provider"""
+        if isinstance(backend, BackendV1):
+            return getattr(backend, "provider", None)
+        elif isinstance(backend, BackendV2):
+            return backend.provider
+        return None
+
+    @staticmethod
+    def qubit_freq_est(backend):
```

Comment:
Perhaps this is good chance to upgrade naming convention. I prefer to use `drive_freqs` and `meas_freqs` (operational aspect), or `qubit_freqs` and `resonator_freqs` (component aspect). Personally I prefer former one, because some hardware architecture may not use resonator for readout. @wshanks any thoughts?

Reply:
Yes, component properties are separately reported as device Hamiltonian parameters (but the context is mixed up due to this confusing name).

Reply:
Done

Reply:
@nkanazawa1989 I think we need to revisit this. `qubit_freq_est` and `meas_freq_est` come from `Backend.defaults()` which is the same for `BackendV1` and `BackendV2`. I think the confusion here was that `defaults` is not in the `Backend` base classes, only added on by the IBM providers (both `qiskit-ibmq-provider` and `qiskit-ibm-provider` add it). Is Pulse an IBM-specific extension? If not, perhaps `defaults` should be added to `BackendV2` with some default return value. Also, perhaps the code in qiskit-experiments that tries to use `defaults` should do a more careful check and give a specific error if `defaults` is not available in the `Backend` class.

By the way, while looking at this, I realized that `defaults` is really Pulse default values (the IBM providers use a `PulseDefaults` type for the return value). Maybe it should be called `pulse_defaults`. Before this, I was always confused about what was in `backend.configuration()` and what was in `backend.defaults()`